### PR TITLE
assorted cppcheck warning fixups

### PR DESCRIPTION
**Pull request by fuzzard**
## Motivation and Context
Fix assorted warnings raised by cppcheck build @lrusak ran recently https://jenkins.kodi.tv/view/Reports/job/LINUX-64-Static-Analysis/lastSuccessfulBuild/cppcheck/

## How Has This Been Tested?
compile/runtime tested osx

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/guilib/GUITexture.h [hunk 1]
```diff
@@ -110,7 +110,7 @@ class CGUITextureBase
   void LoadDiffuseImage();
   bool AllocateOnDemand();
   bool UpdateAnimFrame(unsigned int currentTime);
-  void Render(float left, float top, float bottom, float right, float u1, float v1, float u2, float v2, float u3, float v3);
+  void Render(float left, float top, float right, float bottom, float u1, float v1, float u2, float v2, float u3, float v3)
```
Comment (lrusak) [inline xbmc/guilib/GUITexture.h:113]: whoopsie

File: xbmc/platform/darwin/osx/XBMCHelper.cpp [hunk 2]
```diff
@@ -211,12 +211,12 @@ void XBMCHelper::Configure()
     strConfig += strDelay;
 
     // Find out where we're running from.
-    char real_path[2*MAXPATHLEN];
     char given_path[2*MAXPATHLEN];
     uint32_t path_size = 2*MAXPATHLEN;
 
     if (_NSGetExecutablePath(given_path, &path_size) == 0)
     {
+      char real_path[2*MAXPATHLEN];
```
Comment (ksooo) [inline xbmc/platform/darwin/osx/XBMCHelper.cpp:219]: `2 * MAXPATHLEN` (blank before and after *)

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 3]
```diff
@@ -76,17 +72,18 @@ bool CNetworkInterfaceOsx::GetHostMacAddress(unsigned long host_ip, std::string&
 
   if (sysctl(mib, ARRAY_SIZE(mib), NULL, &needed, NULL, 0) == 0)
   {
-    buf = (char*)malloc(needed);
+    char* buf = (char*)malloc(needed);
```
Comment (ksooo) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp:75]: `static_cast<char*>`

**Review by garbear — APPROVED** [review]

**Review by phunkyfish — APPROVED** [review]

**Review by lrusak — CHANGES_REQUESTED** [review]
You have a couple diffs to apply from jenkins. May I suggest using some clang-format hooks? :wink: https://github.com/barisione/clang-format-hooks

**Comment by fuzzard** [comment]
I'll be honest, i think both of those clang changes are ugly, but I'll do them later today

**Pull request merged**